include/opt.zig

```diff
@@ -0,0 +1,292 @@
+const std = @import("std");
+const assert = std.debug.assert;
+const warn = std.debug.warn;
+const ArrayList = std.ArrayList;
+const Allocator = std.mem.Allocator;
+
+// TODO make tests.  Idk how casting from consts works so no tests for now.
+
+//test "args with no defined flags" {
+//    const args = [_][]const u8{
+//        "test", "arg1",  "--flag1", "arg2",
+//        "--",   "-arg3",
+//    };
+//    var flags = [_]Flag{};
+//    var it = FlagIterator.init(flags[0..0], args[0..]);
+//    assert(std.mem.eql(u8, it.next_arg(), "arg1"));
+//    assert(false);
+//}
+
+const ArgTypeTag = enum {
+    String,
+    None,
+};
+
+const ArgType = union(ArgTypeTag) {
+    String: ?[]u8,
+    None: void,
+};
```

Comment:
It might seem useless to have just String and None flag value types, but later we might want to add Integer or Enum (i.e. the flag value has to be picked from a pre-defined list of allowed values) types, and this way we could do that without having to change any programs that already use this.